### PR TITLE
Concert three letter AA to one letter AA

### LINKED ISSUE #3755: Support three letter AA through API
We should allow user to annotate three letter alteration through annotate/mutations/byProtienChange endpoint

### PR DESCRIPTION
**Pull request by carrotcatnyc**
Fixed #3755 - three letter AA converts to 1 letter AA - not case sensitive as far as I can tell

### CODE REVIEW

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mskcc/cbio/oncokb/model/Alteration.java [hunk 1]
```diff
@@ -150,7 +151,7 @@ public String getAlteration() {
 
 
     public void setAlteration(String alteration) {
-        this.alteration = alteration;
+        this.alteration = AminoAcidConverterUtils.resolveHgvspShortFromHgvsp(alteration);;
```
[the remark this answers is not in the document]
Comment (carrotcatnyc) [reply core/src/main/java/org/mskcc/cbio/oncokb/model/Alteration.java]: Fixed

[the remark this answers is not in the document]
Comment (calvinlu3) [reply core/src/main/java/org/mskcc/cbio/oncokb/model/Alteration.java]: @zhx828 I feel like we should still save the original alteration in the Query object for the end user to have a data point to match response. Maybe we add an hgvsp short field to query?

File: core/src/main/java/org/mskcc/cbio/oncokb/model/Query.java [hunk 2]
```diff
@@ -154,7 +155,7 @@ public void setAlteration(String alteration) {
         if (alteration != null) {
             alteration = alteration.replace("p.", "");
         }
-        this.alteration = alteration;
+        this.alteration = AminoAcidConverterUtils.resolveHgvspShortFromHgvsp(alteration);;
```
Comment (zhx828) [inline core/src/main/java/org/mskcc/cbio/oncokb/model/Query.java:158]: Add some tests in IndicatorUtilsTest to verify three letters variants are supported.

File: core/src/main/java/org/mskcc/cbio/oncokb/util/AminoAcidConverterUtils.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package org.mskcc.cbio.oncokb.util;
+
+public class AminoAcidConverterUtils {
```
Comment (zhx828) [inline core/src/main/java/org/mskcc/cbio/oncokb/util/AminoAcidConverterUtils.java:3]: Add  AminoAcidConverterUtilsTest under the test folder and test all methods under.

**Pull request merged**